### PR TITLE
Timeout fast tests

### LINKED ISSUE #34611: The `Fast Test` script should have timeout for run ~40 minutes and then upload reports to S3
cc @kitaisreal 

Should be added somewhere here https://github.com/ClickHouse/ClickHouse/blob/ddad064285406bb4d91a4cdc7f3a2d1334a8500c/tests/ci/fast_test_check.py#L119

### PR DESCRIPTION
**Pull request by Felixoid**
Changelog category (leave one):
- Build/Testing/Packaging Improvement

Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Set timeout 40 minutes for fast tests.

Fixes https://github.com/ClickHouse/ClickHouse/issues/34611

### CODE REVIEW

**Comment by kitaisreal** [comment]
@Felixoid this should not be in changelog.
```
Set timeout 40 minutes for fast tests, second attempt.
```

File: tests/ci/tee_popen.py [hunk 1]
```diff
@@ -11,23 +16,42 @@
...
         self.command = command
         self.log_file = log_file
         self.env = env
         self.process = None
+        self.timeout = timeout
+
+    def _check_timeout(self):
+        stopwatch = Stopwatch()
+        while stopwatch.duration_seconds < self.timeout:
+            sleep(1)
```
Comment (alesapin) [inline tests/ci/tee_popen.py:29]: Maybe just call sleep for (`self.timeout`)?

**Review by alesapin — APPROVED** [review]

**Comment by alesapin** [comment]
Waiting for FastTest only.

File: tests/ci/tee_popen.py (hunk 1 above)
Comment (Felixoid) [reply tests/ci/tee_popen.py]: yes

**Pull request merged**